src/test/operators/table_scan_sorted_test.cpp

```diff
@@ -0,0 +1,173 @@
+    }
+
+    _table_column_definitions.emplace_back("a", _data_type, nullable);
+
+    auto table = Table::create_dummy_table(_table_column_definitions);
+
+    if (nullable && nulls_first) {
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+    }
+
+    const int table_size = 10;
+    for (int i = 0; i < table_size; i++) {
+      if (_data_type == DataType::Int) {
+        table->append({ascending ? i : table_size - i - 1});
+        if (use_reference_segment) {
+          table->append({ascending ? i : table_size - i - 1});
+        }
+      } else if (_data_type == DataType::String) {
+        table->append({std::to_string(ascending ? i : table_size - i - 1)});
+        if (use_reference_segment) {
+          table->append({std::to_string(ascending ? i : table_size - i - 1)});
+        }
+      } else {
+        Fail("Unsupported DataType");
+      }
+    }
+
+    if (nullable && !nulls_first) {
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+    }
+
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(_encoding_type));
+
+    table->get_chunk(ChunkID(0))->get_segment(ColumnID(0))->set_sort_order(_order_by);
+
+    if (use_reference_segment) {
+      auto pos_list = std::make_shared<PosList>();
+
+      if (nullable && nulls_first) {
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+      }
+
+      for (auto i = 0; i < 2 * table_size; ++i) {
+        if (i % 2 == 0) {
+          pos_list->emplace_back(ChunkID(0), i + (nullable && nulls_first ? 3 : 0));
+        }
+      }
+
+      if (nullable && !nulls_first) {
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+      }
+
+      const auto reference_segment = std::make_shared<ReferenceSegment>(table, ColumnID(0), pos_list);
+
+      auto reference_table = std::make_shared<Table>(_table_column_definitions, TableType::References);
+      reference_table->append_chunk({reference_segment});
+
+      _table_wrapper = std::make_shared<TableWrapper>(std::move(reference_table));
+    } else {
+      _table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+    }
+
+    _table_wrapper->execute();
+  }
+
+  void ASSERT_COLUMN_SORTED_EQ(std::shared_ptr<const Table> table) {
+    size_t i = 0;
+    for (auto chunk_id = ChunkID{0u}; chunk_id < table->chunk_count(); ++chunk_id) {
+      const auto chunk = table->get_chunk(chunk_id);
+
+      for (auto chunk_offset = ChunkOffset{0u}; chunk_offset < chunk->size(); ++chunk_offset, ++i) {
+        const auto& segment = *chunk->get_segment(ColumnID(0));
+
+        const auto found_value = segment[chunk_offset];
+
+        if (found_value == NULL_VALUE) {
+          ASSERT_FALSE(true);
```

Comment:
```suggestion
          FAIL();
```

Reply:
Why no single line `assert(found_value != null_value, "Foobar")`?

Reply:
`ASSERT_EQ` you mean?

Reply:
`ASSERT_NE`.